internal/pubsub/gcppubsub/gcppubsub.go

```diff
@@ -20,35 +20,67 @@ import (
 	"context"
 	"fmt"
 
-	rawgcppubsub "cloud.google.com/go/pubsub/apiv1"
+	raw2 "cloud.google.com/go/pubsub"
+	raw "cloud.google.com/go/pubsub/apiv1"
+	"github.com/google/go-cloud/gcp"
 	"github.com/google/go-cloud/internal/pubsub"
 	"github.com/google/go-cloud/internal/pubsub/driver"
-	pubsubpb "google.golang.org/genproto/googleapis/pubsub/v1"
+	pb "google.golang.org/genproto/googleapis/pubsub/v1"
 )
 
 type topic struct {
 	path   string
-	client *rawgcppubsub.PublisherClient
+	client *raw.PublisherClient
 }
 
-// Close closes the connection to the API service. The user should invoke this
-// when the client is no longer required.
+// TopicOptions will contain configuration for topics.
+type TopicOptions struct{}
+
+// OpenTopic opens the topic on GCP PubSub for the given projectID and
+// topicName. If the topic does not exist then an error will be returned.
+func OpenTopic(ctx context.Context, client *raw.PublisherClient, proj gcp.ProjectID, topicName string, opts *TopicOptions) (*pubsub.Topic, error) {
+	dt, err := openTopic(ctx, client, proj, topicName)
+	if err != nil {
+		return nil, fmt.Errorf("opening topic: %v", err)
```

Comment:
We should prefix error messages with the package, e.g. "gcppubsub: opening topic: ..."

Could be another PR.

Reply:
Done.